components/suggest/src/lib.rs

```diff
@@ -31,6 +31,7 @@ pub struct Suggestion {
     pub icon: Option<Vec<u8>>,
     pub impression_url: Option<String>,
     pub click_url: Option<String>,
+    pub provider: Option<ProviderType>,
```

Comment:
I'm really stepping out of my wheelhouse here, so please feel free to ignore this - but it's not clear this needs to be an option - it looks like you panic when reading the database and find an unknown option. Even if you eventually do not panic, it seems likely you would just ignore an unknown provider - so having this as a non-option might make life easier for consumers if they never have to deal with a `None` in practice.

If you do want to keep it: the DB has the column as non-null, so if you translate an unknown value to `None`, you've lost the actual (unknown) int value - so another option would be another enum variant `Unknown(u8)` - this means you still avoid the `Option<>` while still being able to handle "unknown" values *and* being able to know what the unknown value is (eg, for logging etc).

Reply:
it's just my newbie rust abilities shining through 😬 .

Reply:
> it's just my newbie rust abilities shining through 😬 .

Don't worry, we all remember that feeling very well :) Learning Rust is guaranteed to bring the latent imposter syndrome in everyone to the surface :) It takes longer to feel confident in Rust than learning most other languages in my experience, but it does happen :)